ada_feeding_perception/ada_feeding_perception/table_detection.py

```diff
@@ -0,0 +1,483 @@
+# Standard imports
+import math
+import threading
+from typing import Union
+
+# Third-party imports
+import cv2
+from cv_bridge import CvBridge
+import numpy as np
+import rclpy
+from rclpy.callback_groups import MutuallyExclusiveCallbackGroup
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from sensor_msgs.msg import CameraInfo, CompressedImage, Image
+from std_srvs.srv import SetBool
+
+from scipy.spatial.transform import Rotation as R
+from geometry_msgs.msg import (
+    Pose,
+    PoseStamped,
+    Point,
+    Quaternion,
+)
+
+# Local imports
+from ada_feeding_perception.helpers import (
+    depth_img_to_pointcloud,
+    get_img_msg_type,
+    ros_msg_to_cv2_image,
+)
+
+
+class TableDetectionNode(Node):
+    """
+    This node subscribes to the camera info, depth image, and RGB image topics and exposes a
+    service to toggle table detection on and off. When on, the node publishes
+    a 3D PoseStamped location of the center of the table with respect to the camera's frame of
+    perspective at a specified rate.
+    """
+
+    def __init__(self):
+        """
+        Initialize the TableDetectionNode.
+        """
+
+        super().__init__("table_detection")
+
+        # Table plane parameters
+        self._hough_accum = 1.5
```

Comment:
There is no reason another class would need to use these, right? If so, you should use the double underscore instead of the single one: https://stackoverflow.com/a/1301369